app/components/SearchMap.js

```diff
@@ -1,96 +1,82 @@
 'use strict';
 
 import _ from 'lodash';
+import L from 'leaflet';
 import React from 'react';
-import {Map, Marker, Popup, TileLayer} from 'react-leaflet';
-import {Link} from 'react-router';
 
-import {HELSINKI_COORDINATES} from '../constants/AppConstants';
-import {getBounds, getPosition} from '../core/mapUtils';
+import BaseMap from './BaseMap';
+import {HELSINKI_COORDINATES, MAP_ZOOM} from '../constants/MapConstants';
+import {getMarkerIcon, getPosition} from '../core/mapUtils';
 
-function getMarker(school) {
-  const position = getPosition(school.location);
-  return (
-    <Marker key={school.id} position={position} ref={'marker-' + school.id}>
-      <Popup>
-        <span>
-          <Link params={{schoolId: school.id}} to='school'>{school.name.officialName}</Link>
-        </span>
-      </Popup>
-    </Marker>
-  );
-}
-
-class SearchMap extends React.Component {
+class SearchMap extends BaseMap {
   componentDidMount() {
-    this.map = this.refs.map.getLeafletElement();
-    const coordinates = this.getCoordinates(this.props);
-    this.centerMap(coordinates);
+    this.map = L
+      .map(React.findDOMNode(this.refs.map), this.mapOptions)
+      .setView(HELSINKI_COORDINATES, MAP_ZOOM);
+    L.control.layers(this.layers).addTo(this.map);
+    this.markerGroup.addTo(this.map);
+
+    const self = this;
+    this.map.on('baselayerchange', function() {
+      const centerPoint = self.map.getCenter();
+      const zoom = MAP_ZOOM;
+      self.map.setView(centerPoint, zoom);
+    });
```

Comment:
You could have pass context using "( ) => " notation. That way you can use this inside the anonymous function and you don't need to declare self.